### PR TITLE
Move all test related files into a subdirectory

### PR DESCRIPTION
**Pull request by jhachenbergerSIT**
In order to keep all test-related files within one folder,
this commit creates a new folder 'test/', moves the
test-related files into it and intrdocues a second makefile
within this folder dedicated to run the tests.
The original makefile still contains a 'test' target, but simply
delegates this task to the second makefile.

(fixes david-a-wheeler/flawfinder#11)

Signed-off-by: Julien Hachenberger <julien.hachenberger@sit.fraunhofer.de>

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -133,88 +132,8 @@ time:
...
-test-is-correct: test-results.txt
-	cp -p test-results.txt correct-results.txt
-	cp -p test-results.html correct-results.html
-	cp -p test-results.csv correct-results.csv
-	cp -p test-results-004.txt correct-results-004.txt
-	cp -p test-results-005.txt correct-results-005.txt
-	cp -p test-results-006.txt correct-results-006.txt
-	cp -p test-results-008.txt correct-results-008.txt
+test:
+	$(MAKE) -C $(TESTDIR) test
```
Comment (david-a-wheeler) [inline makefile]: The POSIX standard for make does not include a "-C" option: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/make.html

I think everything is uses just POSIX make capabilities.  I'm not mortally opposed to extensions if needed, but in this case I think you can just use this instead (untested);

~~~~
cd $(TESTDIR); $(MAKE) test
~~~~

You should also pass through "make check" (unless that's already done elsewhere).

File: makefile [hunk 2]
```diff
@@ -100,7 +99,7 @@ distribute: clean flawfinder.pdf flawfinder.ps
 	rm -fr build dist flawfinder.egg-info ,tempdir
 	chmod -R a+rX *
 	mkdir ,tempdir
-	cp -p [a-zA-Z]* ,tempdir
+	cp -r -p [a-zA-Z]* ,tempdir
```
[the remark this answers is not in the document]
Comment (jhachenbergerSIT) [reply makefile]: Sorry, accidentally missed to submit this comment yesterday...

File: makefile (hunk 1 above)
Comment (jhachenbergerSIT) [reply makefile]: Narg, in fact, I did not check if it is POSIX compliant, I will replace it with your proposed solution and also passthrough the `check` target.

Comment (jhachenbergerSIT) [reply makefile]: I will also reintroduce the `test-is-correct` target.

**Comment by jhachenbergerSIT** [comment]
I provided fixup commits to address the issues spotted by you. However, personally, I would prefer to amend the initial commit of this PR in order to leave only working commits. But this probably breaks down to a matter of preference, so it's up to you.

**Comment by david-a-wheeler** [comment]
I prefer to leave original commits as they are, because people can see what changed.

For now -r seems reasonable.

**Comment by david-a-wheeler** [comment]
BTW, we can always do "cp -r ..." and then delete things from ,tempdir if we do NOT want them.  That would avoid failing to include something.

**Pull request merged**